contracts/Core/Constants.sol

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import "./interface/IConstants.sol";
+import "./ACL.sol";
+
+contract Constants is IConstants, ACL {
+
+    uint32 public override commit = 0;
+    uint32 public override reveal = 1;
+    uint32 public override propose = 2;
+    uint32 public override dispute = 3;
+
+    // TODO: Analyze what can be the maximum value (if needs to be reset) of these params.
+    // constant type can be readjusted to some smaller type than uint256 for saving gas (storage variable packing).
+    // penalty not reveal = 0.01% per epch
+    uint256 public override penaltyNotRevealNum = 1;
+    uint256 public override penaltyNotRevealDenom = 10000;
+
+
+    uint256 public override minStake = 100 * (10 ** 18);
+    uint256 public override unstakeLockPeriod = 1;
+    uint256 public override withdrawLockPeriod = 1;
+    uint256 public override maxAltBlocks = 5;
+    uint256 public override epochLength = 40;
+    uint256 public override numStates = 4;
+    uint256 public override exposureDenominator = 1000;
+
+
+    // keccak256("BLOCK_CONFIRMER_ROLE")
+    bytes32 constant private _BLOCK_CONFIRMER_HASH = 0x18797bc7973e1dadee1895be2f1003818e30eae3b0e7a01eb9b2e66f3ea2771f; 
+    
+    // keccak256("JOB_CONFIRMER_ROLE")
+    bytes32 constant private _JOB_CONFIRMER_HASH = 0xbe7b58e17bf6adaa0f209cd0db8b128282fc68a42f2dd649b4d8ea579f1b078f;
+    
+    // keccak256("STAKER_ACTIVITY_UPDATER_ROLE")
+    bytes32 constant private _STAKER_ACTIVITY_UPDATER_HASH = 0x4cd3070aaa07d03ab33731cbabd0cb27eb9e074a9430ad006c96941d71b77ece;
+    
+    // keccak256("STAKE_MODIFIER_ROLE")
+    bytes32 constant private _STAKE_MODIFIER_HASH = 0xdbaaaff2c3744aa215ebd99971829e1c1b728703a0bf252f96685d29011fc804;
+
+    function setCommit(uint32 _commit) external onlyRole(DEFAULT_ADMIN_ROLE) { 
+        commit = _commit;
+    }
+
+    function setReveal(uint32 _reveal) external onlyRole(DEFAULT_ADMIN_ROLE) { 
+        reveal = _reveal;
+    }
+
+    function setPropose(uint32 _propose) external onlyRole(DEFAULT_ADMIN_ROLE) { 
+        propose = _propose;
+    }
+
+    function setDispute(uint32 _dispute) external onlyRole(DEFAULT_ADMIN_ROLE) { 
+        dispute = _dispute;
+    }
+
+    function setPenaltyNotRevealNum(uint256 _penaltyNotRevealNumerator) external onlyRole(DEFAULT_ADMIN_ROLE) { 
+        penaltyNotRevealNum = _penaltyNotRevealNumerator;
+    }
+
+    function setPenaltyNotRevealDeom(uint256 _penaltyNotRevealDenom) external onlyRole(DEFAULT_ADMIN_ROLE) { 
+        penaltyNotRevealDenom = _penaltyNotRevealDenom;
+    }
+
+    function setUnstakeLockPeriod(uint256 _unstakeLockPeriod) external onlyRole(DEFAULT_ADMIN_ROLE) { 
```

Comment:
See rather than ```onlyRole(DEFAULT_ADMIN_ROLE)```
Can we create one role like say "Governer" 
The thing is in near future we want DAO to govern these values, so it would be better if we have a dedicated role.
As of now, we can assign this Governer role to Admin.

Reply:
>  Can we create one role like say "Governer"

Sounds good to me, but for now I avoided it, why? Because currently there is not a clear roadmap how `DAO`  will be implemented, I would rather avoid implementing something that is not quite on the spec yet.

Though I agree with you, `Governer` role can be added in near future.

Reply:
Okay I agree, maybe we should create issue for the same

Reply:
Admin role works for now.